src/app/clusters/chime-server/chime-server.cpp

```diff
@@ -0,0 +1,169 @@
+#include <protocols/interaction_model/StatusCode.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::Chime;
+using namespace chip::app::Clusters::Chime::Attributes;
+using chip::Protocols::InteractionModel::Status;
+
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+ChimeServer::ChimeServer(EndpointId endpointId, ChimeDelegate & delegate) :
+    AttributeAccessInterface(MakeOptional(endpointId), Chime::Id),
+    CommandHandlerInterface(MakeOptional(endpointId), Chime::Id), mDelegate(delegate)
+{
+    mDelegate.SetChimeServer(this);
+}
+
+ChimeServer::~ChimeServer()
+{
+    AttributeAccessInterfaceRegistry::Instance().Unregister(this);
+    CommandHandlerInterfaceRegistry::Instance().UnregisterCommandHandler(this);
+}
+
+
+CHIP_ERROR ChimeServer::Init()
+{
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+    return CHIP_NO_ERROR;
+}
+
+// AttributeAccessInterface
+CHIP_ERROR ChimeServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == Chime::Id);
+    
+    switch (aPath.mAttributeId)
+    {
+    case InstalledChimeSounds::Id:
+        return aEncoder.Encode(mDelegate.GetInstalledChimeSounds());
+
+    case ActiveChimeSoundId::Id:
+        return aEncoder.Encode(mDelegate.GetActiveChimeSoundId());
+
+    case Enabled::Id:
+        return aEncoder.Encode(mDelegate.GetEnabled());
+    }
+    
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ChimeServer::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+{
+    VerifyOrDie(aPath.mClusterId == Chime::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    case ActiveChimeSoundId::Id: {
+        uint8_t newValue;
+        ReturnErrorOnFailure(aDecoder.Decode(newValue));
+        ReturnErrorOnFailure(SetActiveChimeSoundId(newValue));
+        return CHIP_NO_ERROR;
+
+    }
+    case Enabled::Id: {
+        bool newValue;
+        ReturnErrorOnFailure(aDecoder.Decode(newValue));
+        ReturnErrorOnFailure(mDelegate.SetEnabled(newValue));
+        return CHIP_NO_ERROR;
+    }
+        
+    default:
+        // Unknown attribute
+        return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+    }
+
+}
+
+CHIP_ERROR ChimeServer::SetActiveChimeSoundId(uint8_t soundId)
+{
+    uint8_t currentSoundId = mDelegate.GetActiveChimeSoundId();
+    bool activeSoundIdChanged = !(currentSoundId == soundId);
+
+    VerifyOrReturnError(activeSoundIdChanged, CHIP_NO_ERROR);
+    VerifyOrDie(mDelegate.SetActiveChimeSoundId(soundId) == CHIP_NO_ERROR);
+    MatterReportingAttributeChangeCallback(GetEndpointId(), Chime::Id, ActiveChimeSoundId::Id);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ChimeServer::SetEnabled(bool enabled)
```

Comment:
OK, but the write path is not calling this?

Reply:
Thanks, that's an error